src/components/search/SearchDirective.js

```diff
@@ -1,756 +1,259 @@
-              var counter = 0;
-              counter = !hasLocationResults ? counter : counter + 1;
-              counter = !hasFeatureResults ? counter : counter + 1;
-              counter = !scope.hasLayerResults ? counter : counter + 1;
-              for (var cssClass in scope.nbOfSuggestionsLists) {
-                if (cssClass === 'ga-search-' + counter) {
-                  scope.nbOfSuggestionsLists[cssClass] = true;
-                } else {
-                  scope.nbOfSuggestionsLists[cssClass] = false;
+            $scope.key = function(evt) {
+              if (evt.keyCode == 13) {
+                if (evt.target && evt.target.blur) {
+                  evt.target.blur();
                 }
               }
             };
+            // Result set announces their results (put here because we need
+            // element)
+            $scope.childoptions.announceResults = function(type, nr) {
+              $scope.restat.update(type, nr);
 
-            viewDropDown.on('gaSuggestionsRendered', function(evt) {
-              var el;
-              if (viewDropDown.isVisible()) {
-                $timeout(setListCount);
-                el = element.find('.tt-dataset-' + evt.data.name);
-                el.attr('ng-class', 'nbOfSuggestionsLists');
-                $compile(el)(scope);
-                if (el) {
-                  el = el.find('.tt-suggestions');
-                  if (el) {
-                    gaSwisssearch.feed(el);
-                  }
-                  el.scrollTop(0);
-                }
-              }
-              gaSwisssearch.check();
-            });
-
-            scope.clearInput = function() {
-              taElt.val('');
-              taElt.data('ttView').inputView.setQuery('');
-              scope.query = '';
-              gaMarkerOverlay.remove(map);
-              unregisterMove();
-              viewDropDown.clearSuggestions();
-              scope.searchFocused = false;
-              // Display the placeholder
-              if (gaBrowserSniffer.msie < 10) {
+              if (swisssearchActive &&
+                  $scope.restat.done()) {
                 $timeout(function() {
-                  taElt.trigger('blur');
-                }, 0, false);
-              }
-            };
-
-            scope.$on('gaTopicChange', function(event, topic) {
-              currentTopic = topic.id;
-            });
-
-            var triggerSearch = function(dataSetIndex) {
-              if (scope.query !== '') {
-                var dataset = taElt.data('ttView').datasets[dataSetIndex];
-                dataset.getSuggestions(scope.query,
-                    function(suggestions) {
-                      viewDropDown.renderSuggestions(dataset, suggestions,
-                          false);
+                  var renderedResults = element.find('.ga-search-item');
+                  if (renderedResults.length === 1) {
+                    renderedResults.click();
+                  }
```

Comment:
no need of the `if`, jquery handle correctly (nothing happens) if `element.find('.ga-search-item')` returns an empty array.